buildstockbatch/gcp/gcp.py

```diff
@@ -564,6 +564,22 @@ def copy_files_at_cloud(self, files_to_copy):
 
     def start_batch_job(self, batch_info):
         """Implements :func:`DockerBase.start_batch_job`"""
+        # Make sure the default subnet for this region has access to Google APIs when external IP addresses are blocked.
+        subnet_client = compute_v1.SubnetworksClient()
+        op = subnet_client.set_private_ip_google_access(
+            project=self.gcp_project,
+            region=self.region,
+            subnetwork="default",
+            subnetworks_set_private_ip_google_access_request_resource=compute_v1.SubnetworksSetPrivateIpGoogleAccessRequest(  # noqa
+                private_ip_google_access=True
+            ),
+        )
+        op.result()
+        if op.error_code:
+            logger.error(
+                f"Error ({op.error_code}) updating subnet settings to allow private Google access: {op.error_message}"
+            )
+
```

Comment:
Removing the external IP limitation of 128, do we have any similar limitation with private IPs? I am trying to understand how many concurrent simulations can we achieve with eliminating the use of external IPs

Reply:
Looking through the relevant quotas [here](https://cloud.google.com/vpc/docs/quota), I think the relevant ones are high enough (e.g. the `Instances per VPC network` limit is 15,000) that we're more likely to hit limits on general compute resources (e.g. total number of CPUs) first. (Note that we're _not_ using static internal IPs, so those limits don't apply here.)

Most of these quotas also apply per-region, so if needed we could also start running jobs in different regions. (Other regions also have different quotas and prices, so this might be worth looking into anyway.)

Reply:
This great! right and general compute resources have much higher (compared to 128) quota limits, so we should be good on that part then.
Agreed! I think it would be worth looking into quotas and costs for other regions to see if switching to those would make sense